compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -1142,27 +1144,62 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     }
 
     /// Turning a "maybe pointer" into a proper pointer (and some information
-    /// about where it points), or an absolute address.
+    /// about where it points), or an absolute address. Also returns whether
+    /// the pointer had a permissive provenance.
     pub fn ptr_try_get_alloc_id(
         &self,
         ptr: Pointer<Option<M::PointerTag>>,
     ) -> Result<(AllocId, Size, Pointer<M::PointerTag>), u64> {
+    // ) -> (Result<(AllocId, Size, Pointer<M::PointerTag>), u64>, bool) {
+        // match ptr.into_pointer_or_addr() {
+        //     Ok(ptr) => {
+        //         let res = match M::ptr_reify_alloc(self, ptr) {
+        //             Some((alloc_id, offset, ptr)) => Ok((alloc_id, offset, ptr)),
+        //             None => {
+        //                 let (_, addr) = ptr.into_parts(); // address is absolute
+        //                 Err(addr.bytes())
+        //             }
+        //         };
+
+        //         (res, ptr.provenance.is_permissive())
+        //     }
+        //     Err(addr) => (Err(addr.bytes()), false), // `invalid_ptr` / transmute
+        // }
```

Comment:
What happened here? Looks like `ptr_try_get_alloc_id` is just an alias for `ptr_try_reify_alloc` now?